From 145412013241e6db9a997c84405cef7a504e1403 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Wed, 13 Mar 2024 05:59:48 -0400
Subject: [PATCH 01/19] Loosen requirements

* Remove unused libclang requirement
* Remove unecessary tensorflow-io-gcs-filesystem
* unpin numpy

Originally by: Uwe L. Korn <uwe.korn@quantco.com>
Co-authored-by: Mark Harfouche <mark.harfouche@gmail.com>
Co-authored-by: ngam
---
 tensorflow/tools/pip_package/setup.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index cdec9d3a..9c563514 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -89,11 +89,7 @@ REQUIRED_PACKAGES = [
     'gast >=0.2.1,!=0.5.0,!=0.5.1,!=0.5.2',
     'google_pasta >= 0.1.1',
     'h5py >= 3.10.0',
-    'libclang >= 13.0.0',
     'ml_dtypes ~= 0.3.1',
-    # TODO(b/304751256): Adjust the numpy pin to a single version, when ready
-    'numpy >= 1.23.5, < 2.0.0 ; python_version <= "3.11"',
-    'numpy >= 1.26.0, < 2.0.0 ; python_version >= "3.12"',
     'opt_einsum >= 2.3.2',
     'packaging',
     # pylint:disable=line-too-long
@@ -106,8 +102,6 @@ REQUIRED_PACKAGES = [
     'termcolor >= 1.1.0',
     'typing_extensions >= 3.6.6',
     'wrapt >= 1.11.0',
-    # TODO(b/305196096): Remove the <3.12 condition once the pkg is updated
-    'tensorflow-io-gcs-filesystem >= 0.23.1 ; python_version < "3.12"',
     # grpcio does not build correctly on big-endian machines due to lack of
     # BoringSSL support.
     # See https://github.com/tensorflow/tensorflow/issues/17882.
-- 
2.43.2

